### DISCUSSION TITLE
T-Deck-plus / T-deck plus touchscreen unresponsive or laggy: a solution

### DISCUSSION THREAD

**Discussion by mariotti** [meshtastic/firmware#5606] [General]
Hoi all,

I write this to save you from long searches on the net. Let me write few keywords first:

**t-deck, t-deck-plus, t-deck plus, touch, touchscreen, calibration, capacitive.**

Please add your search keys in comments or the like.

The main issue is that the t-deck-plus presents a very laggy, unresponsive behaviour.
Something very difficult to even report, in my case I had it working nicely for first touches.
I started to question if it was an hardware issue.

But here the solution!

Apparently there is an HW (hardware) issue with some T-Deck plus versions. (old?)
This was originally discussed here:

https://github.com/Xinyuan-LilyGO/T-Deck/issues/42

There, it is presented a solution which might not work today.

But if you check this repo: https://github.com/Xinyuan-LilyGO/T-Deck

You will find some niche files here:

https://github.com/Xinyuan-LilyGO/T-Deck/tree/master/firmware

Please read the README for more details.

The basic solution is to flash first one of the fixing firmware, then flash your
preferred firmware.

For Windows users, I hope that the instructions in the README will work. :)
(sorry I cannot check this, not a win user)

For Mac users, I had no command line way to flash the .bin from the repo.
So I used

https://espressif.github.io/esptool-js/

And flashing:

https://github.com/Xinyuan-LilyGO/T-Deck/blob/master/firmware/T-Deck-Plus-TouchFix_241203.bin

This will fix it, and it should work also for Win users.

NOTES:

It is not that obvious on which versions of the t-deck plus this is required.

Please also check: 
https://github.com/meshtastic/firmware/discussions/5432

Q part

Please let me know if this worked.

Please let us know if you have different solutions.

Please let us know if there are related issues.

**Comment by mariotti** [comment]
The idea is to close this discussion asap, and just make it available for the search engines :) But I keep it open for a while just to be sure ...

**Comment by radimsejk** [comment]
Worked for me on Windows without a problem. Thank you very much for this.

**Comment by chrisl8** [comment]
THANK YOU SO MUCH!!!

I thought my brand new T-Deck was trash, but nope, just needed this.

For anyone confused by all of the links and reading, all I did was follow this part of the original post:

> So I used
>
>https://espressif.github.io/esptool-js/
>
>And flashing:
>
>https://github.com/Xinyuan-LilyGO/T-Deck/blob/master/firmware/T-Deck-Plus-TouchFix_241203.bin

NOTE that you need to get the unit into "flash" mode first. You can use the Meshtastic Web Flasher's "1200bps Reset" to do this. When you pick the Serial Port in the browser, it should have the word "jtag" and not the word "Espressif" in it. That is how you know you got it into flash mode.

The one thing on this page https://github.com/Xinyuan-LilyGO/T-Deck/tree/master/firmware that is helpful is under "Use Web Flasher" there is a gif showing how to use the espressif.github.io web flasher above to flash the file.

After you flash that file with the web flasher, go back and flash the Meshatastic firmware and your touchscreen should work beautifully!

NOTE: Some people have reported that you cannot erase and install you just have to install after the touch screen bin file fix. If you erase and install it will remove the fix.

Ask around on the Meshtastic Discord if you get stuck and feel free to PING me here as well if I need to write up better instructions for anyone.

UPDATE: Can confirm that if you do an ERASE and Flash from the Meshtastic Web Flasher you will have to do this again.

ALSO, if your touchscreen responds but seems misaligned, find the "Screen Calibration" under Settings.

Here are some screenshots from using the ESP Tool flasher:

1. You have to Connect before you can do anything.
<img width="371" height="396" alt="image" src="https://github.com/user-attachments/assets/77325d0a-e6ed-4036-9e48-b9ae111d3e96" />

If the option is "Espressif..." it won't work, you need to get into "flash mode"

BAD:
<img width="274" height="33" alt="image" src="https://github.com/user-attachments/assets/31a06bfa-b67d-42fd-b548-0e25f5784257" />

 - Turn the device OFF
 - Press and hold the center trackball
 - Connect the USB-C cable
 - Toggle the switch to ON
 - Count to ten
 - Release the center trackball

GOOD:
<img width="274" height="33" alt="image" src="https://github.com/user-attachments/assets/63d2d1dd-7bbd-4159-8e63-e622ab213fb5" />

**Comment by Nukleari** [comment]
Worked for me as well, ordered my T-Deck Plus on may 10 from the German warehouse. I was confused about the ui at first as I wasn't really able to short press icons 90% of the time and experienced unresponsive touch behavior overall. After the fix touch seems to work perfectly.

**Comment by Langoor2** [comment]
Also have a t-deck from German warehouse with the same problem, but this does not seem to work with the newest meshtastic alpha release...

**Comment by evenstephen** [comment]
Just applied this on my T-Deck Plus.  Wow, what an improvement!  Amazon order last month, I don't know which warehouse LilyGo shipped from.  Anyhow, worked great!  I don't know if it is important, but the [GIF](https://github.com/Xinyuan-LilyGO/T-Deck/tree/master/firmware) shows changing the 0x100 default location to 0 in the esptool flasher.  Missed that step the first time.  Also to remember how to enter DFU mode on the T-Deck (hold down the trackball, then press the side reset button, release both).
